packages/cactus-test-tooling/src/main/typescript/iroha/iroha-test-ledger.ts

```diff
@@ -0,0 +1,352 @@
+);
+
+export class IrohaTestLedger implements ITestLedger {
+  public readonly imageVersion: string;
+  public readonly imageName: string;
+  public readonly rpcToriiPort: number;
+  public readonly envVars: string[];
+  public readonly emitContainerLogs: boolean;
+  public readonly postgresHost: string;
+  public readonly postgresPort: number;
+
+  private readonly log: Logger;
+  private container: Container | undefined;
+  private containerId: string | undefined;
+
+  constructor(public readonly options: IIrohaTestLedgerOptions) {
+    const fnTag = `IrohaTestLedger#constructor()`;
+    if (!options) {
+      throw new TypeError(`IrohaTestLedger#ctor options was falsy.`);
+    }
+    Checks.nonBlankString(options.postgresHost, `${fnTag} postgresHost`);
+    Checks.truthy(options.postgresPort, `${fnTag} postgresPort`);
+
+    this.postgresHost = options.postgresHost;
+    this.postgresPort = options.postgresPort;
+
+    this.imageVersion =
+      options.imageVersion || IROHA_TEST_LEDGER_DEFAULT_OPTIONS.imageVersion;
+    this.imageName =
+      options.imageName || IROHA_TEST_LEDGER_DEFAULT_OPTIONS.imageName;
+    this.rpcToriiPort =
+      options.rpcToriiPort || IROHA_TEST_LEDGER_DEFAULT_OPTIONS.rpcToriiPort;
+    this.envVars = options.envVars || [
+      ...IROHA_TEST_LEDGER_DEFAULT_OPTIONS.envVars,
+    ];
+
+    this.envVars.push(`IROHA_POSTGRES_HOST=${this.postgresHost}`);
+    this.envVars.push(`IROHA_POSTGRES_PORT=${this.postgresPort}`);
+
+    this.emitContainerLogs = Bools.isBooleanStrict(options.emitContainerLogs)
+      ? (options.emitContainerLogs as boolean)
+      : true;
+
+    this.validateConstructorOptions();
+    const label = "iroha-test-ledger";
+    const level = options.logLevel || "INFO";
+    this.log = LoggerProvider.getOrCreate({ level, label });
+  }
+
+  public getContainer(): Container {
+    const fnTag = "IrohaTestLedger#getContainer()";
+    if (!this.container) {
+      throw new Error(`${fnTag} container not yet started by this instance.`);
+    } else {
+      return this.container;
+    }
+  }
+
+  public get imageFqn(): string {
+    return `${this.imageName}:${this.imageVersion}`;
+  }
+
+  public async getRpcToriiPortHost(): Promise<string> {
+    const ipAddress = "127.0.0.1";
+    const hostPort: number = await this.getRpcToriiPort();
+    return `http://${ipAddress}:${hostPort}`;
+  }
+
+  /**
+   * Output is based on the standard Iroha genesis.block contents.
+   *
+   * @see https://github.com/hyperledger/iroha/blob/main/example/genesis.block
+   */
+  public getGenesisAdminAccount(): string {
+    return "admin@test";
+  }
+
+  /**
+   * Output is based on the standard Iroha admin user public key.
+   *
+   * @see https://github.com/hyperledger/iroha/blob/main/example/admin%40test.pub
+   * @see https://github.com/hyperledger/iroha/blob/main/example/genesis.block
+   */
+  public getGenesisAccountPubKey(): string {
+    return "313a07e6384776ed95447710d15e59148473ccfc052a681317a72a69f2a49910";
```

Comment:
I suggest not to hardcode the public key, but generate its from private key. There should be function to do the job.
I found:
```
cryptoHelper.derivePublicKey(privateKeyHex)
```
in `src/cryptoHelper.ts`

Reply:
Thank you for pointing out! 
How did I address it: I pulled the file from the running container so that I can retrieve the string.
But thank you for bringing up this `src/cryptoHelper.ts` file, I used it in other places to generate user/node key pairs.